### PR TITLE
rfc(taskworker): First pass at storing inflight tasks in redis

### LINKED ISSUE #78697: Build prototype of pending tasks with Redis
Use redis as the backing store for pending tasks. Figure out what keys we will need and how to manage state across those keys.

### PR DESCRIPTION
**Pull request by enochtangg**
This PR is a first pass at implementing the PendingTaskStore "interface" with redis backed storage. The docstring in the `InflightTaskStoreRedis` class contains a description of how keys are structured, their pros and cons, and next steps. The basic functionalities have been validate through manual testing, though I'm sure there are a bunch of bugs. This PR is meant to be used to gather some initial numbers from the test harness. From there, we can iterate on the key structure and implementation.

### CODE REVIEW

File: src/sentry/taskworker/pending_task_store.py [hunk 1]
```diff
@@ -272,6 +277,433 @@ def remove_completed(self) -> None:
...
+            self._set_task(task_id, task_json)
+            del inflight_dict[
+                "activation"
+            ]  # remove the activation field as it is stored in a separate key
+            self._set_inflight_activation_attributes(task_id, inflight_dict)
+            self._set_inflight_activations(status, offset, task_id)
+
+    def get_pending_task(self) -> InflightActivation | None:
+        # TODO: can maybe use zpopmin instead here?
+        task_id = self._get_min_offset_activation(TASK_ACTIVATION_STATUS_PENDING)
```
Comment (markstory) [inline src/sentry/taskworker/pending_task_store.py:473]: zpopmin could help races and duplicate task execution if that shows up in the test harness.

File: src/sentry/taskworker/pending_task_store.py [hunk 2]
```diff
@@ -272,6 +277,433 @@ def remove_completed(self) -> None:
...
+        if task_id is None:
+            return None
+
+        offset = self._get_inflight_activation_field(task_id, "offset")
+
+        deadline = datetime.now() + timedelta(seconds=30)
+        formatted_deadline = (
+            deadline.isoformat() + "Z"
+        )  # TODO: timestamps in task JSON are encoded as a string in RGC 3339 format "{year}-{month}-{day}T{hour}:{min}:{sec}[.{frac_sec}]Z"
```
Comment (markstory) [inline src/sentry/taskworker/pending_task_store.py:482]:
```suggestion
        )  # TODO: timestamps in task JSON are encoded as a string in RFC 3339 format "{year}-{month}-{day}T{hour}:{min}:{sec}[.{frac_sec}]Z"
```

**Review by markstory — COMMENTED** [review]
Nice work :clap: I'm interested to see how it compares in throughput to other storage options we're considering.

**Comment by enochtangg** [comment]
Captured comments into [redis prototype](https://www.notion.so/sentry/redis-prototype-11a8b10e4b5d8036a095db92db09fae6) doc

**Pull request merged**